aio_pika/tools.py

```diff
@@ -2,6 +2,38 @@
 from functools import partial
 
 
+def iscoroutinepartial(fn):
```

Comment:
Problem is https://gist.github.com/hellysmile/54412a3154892bbd751679b2f68e9a94 , which means it is no possible to detect `partial` as `asyncio.iscoroutinefunction`. This helper detects what to deal with different type of callback's: `yield from` for coroutines or `loop.call_soon` for regular functions like `asyncio.Queue.put_nowait`